### PR TITLE
Add DOI to README & CITATION

### PR DESCRIPTION
**Pull request by Bisaloo**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -111,20 +111,20 @@ linelist <- sim_linelist(
...
+#> 4  4              Andrew Ho confirmed      m  72 2023-01-05           <NA>
+#> 5  5         Deanna Wallace suspected      f   7 2023-01-05           <NA>
+#> 6  6 Abdur Rasheed el-Aboud suspected      m  18 2023-01-05           <NA>
+#>   date_death date_first_contact date_last_contact ct_value
+#> 1       <NA>               <NA>              <NA>       NA
+#> 2       <NA>         2023-01-04        2023-01-04       NA
+#> 3       <NA>         2023-01-03        2023-01-04       NA
+#> 4       <NA>         2023-01-01        2023-01-05     22.7
+#> 5       <NA>         2023-01-09        2023-01-12       NA
+#> 6       <NA>         2023-01-06        2023-01-07       NA
```
Comment (Bisaloo) [inline README.md:127]: @joshwlambert can we set the seed in this README to avoid unnecessary diffs each time the README is re-rendered?

Comment (joshwlambert) [reply README.md:127]: Yes, I don't see a reason why we cannot set a seed. Please let me know if you're still working on this PR, if not I'll commit the changes and merge.

Comment (Bisaloo) [reply README.md:127]: I'm good with this version. I'm simply wondering if we should keep the URL in the citation or if the DOI is enough :thinking:

Comment (Bisaloo) [reply README.md:127]: Maybe we can default to current year if the `Date` field does not exist?

File: inst/CITATION [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+        !(is.null(e$given) && is.null(e$family)) && !is.na(match("aut",
+            e$role))
+    }, aar)
+}
+
+bibentry(
+  bibtype = "Manual",
+  title   = paste0(meta$Package, ": ", gsub("[[:space:]]+", " ", meta$Title)),
+  author  = author,
+  year    = format(meta$Date, "%Y"),
```
Comment (joshwlambert) [inline inst/CITATION:15]: Would this be a suitable short-term fix? `format(Sys.Date(), "%Y")`

File: README.md (hunk 1 above)
Comment (joshwlambert) [reply README.md:127]: I've commented below on a possible solution.

File: inst/CITATION (hunk 2 above)
Comment (Bisaloo) [reply inst/CITATION:15]: No, because we want to use the date field if present.

**Comment by joshwlambert** [comment]
The most recent commit seems to have removed the `preferred-citation:` field from the `CITATION.cff`, was this the intention @Bisaloo?

File: README.md (hunk 1 above)
Comment (joshwlambert) [reply README.md:127]: Seed issue logged in #57.

**Review by joshwlambert — APPROVED** [review]

**Comment by joshwlambert** [comment]
Given the high proportion of GHA commits to manual commits, I am going to squash merge this PR once all the checks pass.

**Pull request merged**